src/crypto/CHIPCryptoPALOpenSSL.cpp

```diff
@@ -1800,14 +1801,14 @@ CHIP_ERROR ExtractVIDFromX509Cert(const ByteSpan & certificate, VendorId & vid)
         VerifyOrExit(object != nullptr, err = CHIP_ERROR_INTERNAL);
         VerifyOrExit(OBJ_obj2txt(buff, sizeof(buff), object, 0) != 0, err = CHIP_ERROR_INTERNAL);
 
-        if (strncmp(vidNeedle, buff, vidNeedleSize) == 0)
+        if (strncmp(isVID ? vidNeedle : pidNeedle, buff, needleSize) == 0)
         {
             ASN1_STRING * data_entry = X509_NAME_ENTRY_get_data(name_entry);
             VerifyOrExit(data_entry != nullptr, err = CHIP_ERROR_INTERNAL);
             unsigned char * str = ASN1_STRING_data(data_entry);
             VerifyOrExit(str != nullptr, err = CHIP_ERROR_INTERNAL);
 
-            vid = static_cast<VendorId>(strtoul(reinterpret_cast<const char *>(str), NULL, 16));
+            id = static_cast<uint16_t>(strtoul(reinterpret_cast<const char *>(str), NULL, 16));
```

Comment:
```suggestion
            VerifyOrExit(ParseInt(reinterpret_cast<const char *>(str), id, 16));
```

The current implementation doesn't check whether the value returned by `strtoul()` can be casted to `uint16_t`. `ParseInt()` does that.